WordPress/src/main/java/org/xmlrpc/android/XMLRPCUtils.java

```diff
@@ -313,13 +313,25 @@ private static String discoverSelfHostedXmlrpcUrl(String siteUrl, String httpUse
         // add the url as provided by the user
         urlsToTry.add(siteUrl);
 
-        // add a sanitized version of the url
-        final String sanitizedURL = sanitizeSiteUrl(siteUrl);
-        if (!urlsToTry.contains(sanitizedURL)) {
-            urlsToTry.add(sanitizedURL);
+        // add a sanitized version of the http url (if the user didn't specify it)
+        final String sanitizedHttpsURL = sanitizeSiteUrl(siteUrl, true);
+        if (!urlsToTry.contains(sanitizedHttpsURL)) {
+            urlsToTry.add(sanitizedHttpsURL);
+        }
+
+        String appendedXmlrpcUrl = appendXMLRPCPath(sanitizedHttpsURL);
+        if (!urlsToTry.contains(appendedXmlrpcUrl)) {
+            appendedXmlrpcUrl += "?rsd";
```

Comment:
@daniloercoli not very sure I understand what you want me to confirm here. Let me try to though:
So, the `+= "?rsd";` part was actually taken from the old `smartURLCleanerForRSD()` (see here for example: https://github.com/wordpress-mobile/WordPress-Android/blob/fcf063d2db80d6d9dc5c5b7d3769a6de00526ed0/WordPress/src/main/java/org/wordpress/android/ui/accounts/helpers/FetchBlogListWPOrg.java#L229) and is indeed part of the "discovery" process. Not sure I covered you here. Let me know if not! Thanks!